### PR TITLE
Docs: Revised basic tutorials

### LINKED ISSUE #569: Revise tutorial
**Requirements:**
- #516 
- #573 
- #575 

**Tasks:**
- [x] update terminology 
-- Depends on #516
- [x] make sure it uses new user dict functionality; see docstring of StateDict() in https://github.com/AutoResearch/autora-core/blob/main/src/autora/state.py 
-- Depends on #575
- [x] explain why theorist does have a different decorator and wrapper "estimator_on_state" (for theorist we are using sklearn interface; this is different from the interface we use for experiment runners and experimentalists; thus you have to wrap theorists in a different way)
- [x] mention that theorists return sometimes lists and sometimes models (make clear what happens when; related to 
https://github.com/orgs/AutoResearch/discussions/543) [this issue should be covered. Theorists can return both a singular model or a list of models; both are used to update the list of models in the state.
- [x] There are some tutorials that aren't relevant anymore, such as. Would be necessary to remove the ones out of date.

### PR DESCRIPTION
**Pull request by chadcwilliams**
# Description

Completed the basic tutorials with the latest AutoRA version to be released.

*If the issue is on Github, simply link to it using the '#' symbol; otherwise, provide a notion page link)*:
- resolves #569 

# Type of change

*Delete as appropriate:*
- **docs**: Documentation only changes

### CODE REVIEW

File: docs/tutorials/basic/Tutorial-III-Functional-Workflow.ipynb
[the remark this answers is not in the document]
Comment (chadcwilliams) [reply docs/tutorials/basic/Tutorial-III-Functional-Workflow.ipynb:1]: Ah good point! I was about to add this, but really, shouldn't this repo be part of the autora-synthetic repo? @younesStrittmatter

**Comment by benwandrew** [comment]
small point, but per our code style guidelines, notebooks should be "[Camel case with spaces for Jupyter notebooks: Example Notebook.ipynb](https://autoresearch.github.io/autora/contribute/#:~:text=Camel%20case%20with%20spaces%20for%20Jupyter%20notebooks%3A%20Example%20Notebook.ipynb)"

ie, no dashes

File: docs/tutorials/basic/Tutorial-III-Functional-Workflow.ipynb
[the remark this answers is not in the document]
Comment (chadcwilliams) [reply docs/tutorials/basic/Tutorial-III-Functional-Workflow.ipynb:1]: Added to tutorial 3/4

!pip install -q "autora-synthetic-abstract-equation"

But this does not follow our normal install format.

[the remark this answers is not in the document]
Comment (younesStrittmatter) [reply docs/tutorials/basic/Tutorial-III-Functional-Workflow.ipynb:1]: You are right. Let's at least add it to the optional dependencies for now. There will be another abstract package called ssm for using hssm simulations as synthetic runners, so I guess we can use the same logic as for experimentalists and theorists. I'll add a pull-request to change the pyproject.toml file.

**Review by hollandjg — APPROVED** [review]
Looks good to me!

**Pull request merged**